test/RouteTest.php

```diff
@@ -281,25 +284,59 @@ public function testConstructorRaisesExceptionForInvalidMiddleware($middleware)
 
     public function validNonInteropMiddleware()
     {
-        yield 'string-callable' => ['sprintf'];
-        yield 'array' => [['sprintf']];
-        yield 'array-callable' => [[$this, 'setUp']];
+        yield 'string-callable' => ['sprintf', '"callable:sprintf"'];
+        yield 'array' => [['sprintf'], '"array"'];
+        yield 'array-callable' => [[$this, 'setUp'], '"callable"'];
     }
 
     /**
      * @dataProvider validNonInteropMiddleware
      * @param mixed $middleware
+     * @param string $expectedMessage
      */
-    public function testPassingNonInteropMiddlewareToConstructorTriggersDeprecationNotice($middleware)
-    {
-        $spy = (object) ['found' => false];
+    public function testPassingNonInteropMiddlewareToConstructorTriggersDeprecationNotice(
+        $middleware,
+        $expectedMessage
+    ) {
+        $spy = (object) ['found' => false, 'message' => ''];
         $this->errorHandler = set_error_handler(function ($errno, $errstr) use ($spy) {
             $spy->found = true;
+            $spy->message = $errstr;
             return true;
         }, E_USER_DEPRECATED);
 
         new Route('/test', $middleware);
 
         $this->assertTrue($spy->found);
+        $this->assertContains($expectedMessage, $spy->message);
+    }
+
+    public function testNoDeprecationNoticeRaisedForValidHttpInteropMiddlewareArguments()
+    {
+        switch (true) {
+            case (interface_exists(LegacyPre041MiddlewareInterface::class)):
+                $class = LegacyPre041MiddlewareInterface::class;
+                break;
+            case (interface_exists(Legacy041MiddlewareInterface::class)):
+                $class = Legacy041MiddlewareInterface::class;
+                break;
+            case (interface_exists(Legacy050MiddlewareInterface::class)):
+                $class = Legacy050MiddlewareInterface::class;
+                break;
+            default:
+                $this->fail('No http-interop middleware interfaces detected? Did you run composer install?');
+        }
+        $middleware = $this->prophesize($class)->reveal();
```

Comment:
As we proof before we can use just: `Webimpress\HttpMiddlewareCompatibility\MiddlewareInterface` here.

Reply:
I want to be explicit here about what we're testing. The report that prompted these tests was due to _not_ using the polyfill, and supposedly triggering the deprecation notice; this ensures that in that situation, they do _not_ trigger it.

Reply:
Fine, but it's exactly the same and less complicated using polyfill. But if you really want to be explicit ... ok ;-)